File: macos/Scanner/Sources/Scanner/Models/ReportSummary.swift
public enum OverallStatus: String, Codable, Sendable {
    case critical
    case high
    case medium
    case low
    case clean
}

public struct ReportSummary: Equatable, Codable, Sendable {
    public let toolsFound: Int
    public let mcpServersFound: Int
    public let critical: Int
    public let high: Int
    public let medium: Int
    public let low: Int
    public let status: OverallStatus

    public init(
        toolsFound: Int,
        mcpServersFound: Int,
        critical: Int,
        high: Int,
        medium: Int,
        low: Int,
        status: OverallStatus
    ) {
        self.toolsFound = toolsFound
        self.mcpServersFound = mcpServersFound
        self.critical = critical
        self.high = high
        self.medium = medium
        self.low = low
        self.status = status
    }

    public init(scanResult: ScanResult) {
        let findings = scanResult.findings
        let appsWithFacts = Set(
            scanResult.facts.mcpServers.map(\.appId) +
                scanResult.facts.settings.map(\.appId) +
                scanResult.facts.authFiles.map(\.appId) +
                scanResult.facts.extensions.map(\.appId) +
                scanResult.facts.configFiles.map(\.appId) +
                scanResult.facts.appInstallations.filter(\.installed).map(\.appId)
        )

        let critical = findings.filter { $0.severity == .critical }.count
        let high = findings.filter { $0.severity == .high }.count
        let medium = findings.filter { $0.severity == .medium }.count
        let low = findings.filter { $0.severity == .low }.count

        self.init(
            toolsFound: appsWithFacts.count,
            mcpServersFound: scanResult.facts.mcpServers.count,
            critical: critical,
            high: high,
            medium: medium,
            low: low,
            status: ReportSummary.status(critical: critical, high: high, medium: medium, low: low)
        )
    }

    private static func status(critical: Int, high: Int, medium: Int, low: Int) -> OverallStatus {
        if critical > 0 {
            return .critical
        }
        if high > 0 {
            return .high
        }
        if medium > 0 {
            return .medium
        }
        if low > 0 {
            return .low
        }
        return .clean
    }
}

File: windows/Scanner/Models/ReportSummary.cs
namespace AIExposureScanner.Scanner.Models;

public enum OverallStatus
{
    Critical,
    High,
    Medium,
    Low,
    Clean
}

public static class OverallStatusExtensions
{
    public static string ToJsonValue(this OverallStatus status) =>
        status switch
        {
            OverallStatus.Critical => "critical",
            OverallStatus.High => "high",
            OverallStatus.Medium => "medium",
            OverallStatus.Low => "low",
            OverallStatus.Clean => "clean",
            _ => throw new ArgumentOutOfRangeException(nameof(status), status, null)
        };
}

public sealed record ReportSummary(
    int ToolsFound,
    int McpServersFound,
    int Critical,
    int High,
    int Medium,
    int Low,
    OverallStatus Status
)
{
    public static ReportSummary FromScanResult(ScanResult scanResult)
    {
        var appsWithFacts = scanResult.Facts.McpServers.Select(fact => fact.AppId)
            .Concat(scanResult.Facts.Settings.Select(fact => fact.AppId))
            .Concat(scanResult.Facts.AuthFiles.Select(fact => fact.AppId))
            .Concat(scanResult.Facts.Extensions.Select(fact => fact.AppId))
            .Concat(scanResult.Facts.ConfigFiles.Select(fact => fact.AppId))
            .Concat(scanResult.Facts.AppInstallations.Where(fact => fact.Installed).Select(fact => fact.AppId))
            .Distinct(StringComparer.Ordinal)
            .Count();

        var critical = scanResult.Findings.Count(finding => finding.Severity == Severity.Critical);
        var high = scanResult.Findings.Count(finding => finding.Severity == Severity.High);
        var medium = scanResult.Findings.Count(finding => finding.Severity == Severity.Medium);
        var low = scanResult.Findings.Count(finding => finding.Severity == Severity.Low);

        return new ReportSummary(
            appsWithFacts,
            scanResult.Facts.McpServers.Count,
            critical,
            high,
            medium,
            low,
            StatusFromCounts(critical, high, medium, low)
        );
    }

    private static OverallStatus StatusFromCounts(int critical, int high, int medium, int low)
    {
        if (critical > 0)
        {
            return OverallStatus.Critical;
        }
        if (high > 0)
        {
            return OverallStatus.High;
        }
        if (medium > 0)
        {
            return OverallStatus.Medium;
        }
        if (low > 0)
        {
            return OverallStatus.Low;
        }
        return OverallStatus.Clean;
    }
}
